src/wp-includes/rest-api/endpoints/class-wp-rest-autosaves-controller.php

```diff
@@ -220,7 +220,14 @@ public function create_item( $request ) {
 		$prepared_post->ID = $post->ID;
 		$user_id           = get_current_user_id();
 
-		if ( ( 'draft' === $post->post_status || 'auto-draft' === $post->post_status ) && $post->post_author == $user_id ) {
+		// We need to check post lock to ensure the original author didn't leave their browser tab open.
+		if ( ! function_exists( 'wp_check_post_lock' ) ) {
+			require_once( ABSPATH . 'wp-admin/includes/post.php' );
+		}
+
+		$post_lock = wp_check_post_lock( $post->ID );
+
+		if ( ( 'draft' === $post->post_status || 'auto-draft' === $post->post_status ) && (int) $post->post_author === (int) $user_id && false === $post_lock ) {
```

Comment:
Right now this is checking that there isn't an existing post lock at all. I think we also want to allow the post to be updated if the `get_current_user_id` matches the user returned by `wp_check_post_lock`.